### PR TITLE
fix(utils): add support for PAT based remote URLs

### PR DESCRIPTION
**Pull request by burntcarrot**
The current method of extracting owners from remote URLs doesn't work with remote URLs having a PAT in it, for example:

`https://username:<access_token>@github.com/username/repo`

This can be improved by using a regular expression like [this](https://regex101.com/r/xcIeR2/1), which can extract owner names from both type of remote URLs, as shown [here](https://regex101.com/r/xcIeR2/1).

Signed-off-by: burntcarrot <aadhav.n1@gmail.com>

### CODE REVIEW

File: command/repo/status/status.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func (opts *RepoStatusOptions) Run() (err error) {
 	if statusResponse.Activated {
 		pterm.Info.Println("Analysis active on DeepSource (deepsource.io)")
 	} else {
-		pterm.Info.Println("DeepSource analysis is currently not actived on this repository.")
+		pterm.Info.Println("DeepSource analysis is currently not activated on this repository.")
```
Comment (siddhant-deepsource) [inline command/repo/status/status.go:77]: Nice catch 😅

File: utils/fetch_remote.go [hunk 2]
```diff
@@ -87,13 +86,9 @@ func ListRemotes() (map[string][]string, error) {
...
-			u, err := url.Parse(remoteURL)
-			if err != nil {
-				continue
-			}
-			splitPath := strings.Split(u.Path, "/")
-
-			owner = splitPath[0]
+			remoteRegexp := regexp.MustCompile(`.+//(.+)/(.+)/(.+)(\.git)?$`)
+			matched := remoteRegexp.FindStringSubmatch(remoteURL)
+			owner = matched[2]
```
Comment (siddhant-deepsource) [inline utils/fetch_remote.go:91]: @burntcarrot Thanks for this. The `ListRemotes` function is critical for several commands in the CLI which require the repository details. Would be great if you can somehow add unit tests for the remotes extraction logic. Feel free to refactor the function and use dummy input of `git` commands in order to make it easy for writing tests.

**Review by siddhant-deepsource — COMMENTED** [review]
@burntcarrot Good catch and thanks for the patch. Would be great if you can add tests. 🚀

File: utils/fetch_remote.go (hunk 2 above)
Comment (burntcarrot) [reply utils/fetch_remote.go:94]: Added unit tests for the remote extraction logic. Had to refactor some code too. 😓

**Review by siddhant-deepsource — APPROVED** [review]
Great work @burntcarrot!! Let's go 🚀

**Pull request merged**